main.go

```diff
@@ -38,6 +38,7 @@ import (
 
 	ptpv1 "github.com/openshift/ptp-operator/api/v1"
 	"github.com/openshift/ptp-operator/controllers"
+	"github.com/openshift/ptp-operator/pkg/leaderelection"
```

Comment:
Is the leaderelection implementation specific to ptp-operator? Why aren't you just importing https://github.com/openshift/library-go/blob/master/pkg/config/leaderelection/leaderelection.go here?

Reply:
Hey @josephdrichard You're right this isn't specific to ptp-operator. When the PR was opened there was a breaking dependency change between the k8s-23 libs and controller-runtime. It was causing some problems and I figured a bit of code was the better approach for now since library-go wasn't being used else where in the code. Luckily, controller-runtime was updated a few hours so we can import it now.

Reply:
Are you still looking to update this?

Reply:
Apologies, forgot to mention in the reply. This PR has been updated with the latest library-go code and I'm using those methods to drive the leaderelection configs